chrispinkney/cinesync
# Issue-114: Update theme to revise colours and add light theme

## Issue

Closes #114 

## Type of Change



## Description

Theming updates to improve appearance and add light mode:

- Moved `theme.ts` contents to `ThemeRegistry.tsx` in order to use media query to specify light/dark mode user preference.
- Updated theme colours (using a green as primary and blue as secondary, also updated success/warning/error/info colours and action hover colour).
- Removed drop shadow from `cinesync-logo-light.svg` to fix weird white border that would appear between text and shadow
- Changed imdb from image to MUI `<SvgIcon>` to improve functionality with colour changes.
- Updated theming in auth components and about page. Converted css styles into MUI props to ensure theme is used correctly.
- Added hover effect to app logo to give user feedback that it is clickable.
- Added colours to hover effect of delete/share/imdb buttons.
- Added scrollbar styling based on theme colours. Needed to tweak content container height to get scrollbar to render inside content and scale to correct size responsively.
- Added dark/light mode toggle on app header. This will toggle a mode context provided from within the `ThemeRegistry.tsx` file.
- Setup local storage saving and recalling of selected theme mode. If no value in local storage, a media query will be used instead to determine the default mode.

![image](https://github.com/chrispinkney/cinesync/assets/75338069/c019a37c-6b72-4673-8dcf-9cccf31dc7c4)

## Testing the PR

- Verify that the first time the page loads, it loads with the mode specified by your system.
- Toggle light/dark modes on and off and refresh the page to ensure the mode persists.
- Using both light & dark modes, go through all pages of the app to ensure colours and contrasts are okay.

## Checklist
